pages/api/transactions/index.js

```diff
@@ -8,6 +8,19 @@ export default async function handler(request, response) {
       const transactions = await Transaction.find();
       return response.status(200).json(transactions);
     }
+
+    if (request.method === "POST") {
+      try {
+        const transactionData = request.body;
+        const transaction = new Transaction(transactionData);
+        await transaction.save();
+        return response.status(201).json({ status: "Transaction added." });
+      } catch (error) {
+        console.error(error);
+        return response.status(400).json({ error: error.message });
+      }
+    }
+    
   } catch (error) {
```

Comment:
you have two nested try catch blocks, which is a bit overkill. Please refactor such that each request type has its own try catch block.